test/serializer/pure-functions/BinaryExpressions.js

```diff
@@ -0,0 +1,28 @@
+// additional functions
+// abstract effects
+
+let obj1 = global.__abstract ? __abstract({}, '({foo: {valueOf() { return 42; }}})') : {foo: {valueOf() { return 42; }}};
+let obj2 = global.__abstract ? __abstract({}, '({foo: {bar: {valueOf() { return 42; }}}})') : {foo: {bar: {valueOf() { return 42; }}}};
+
+if (global.__makePartial) {
+  __makePartial(obj1);
+  __makePartial(obj2);
+}
+if (global.__makeSimple) {
+  __makeSimple(obj1);
+  __makeSimple(obj2);
+}
```

Comment:
Same comment as below, I think we can lose it if we change `__abstract({}` to `__abstract('object'`. Simple/partial doesn't matter here.